source/packages/modulo_core/include/modulo_core/Cell.hpp

```diff
@@ -478,7 +476,7 @@ Cell::Cell(const std::string& node_name, const std::chrono::duration<int64_t, Du
                                                                                                                               period_(period) {
   // add the update parameters call
   std::function<void(void)> update_parameters_fnc = std::bind(&Cell::update_parameters, this);
-  this->parameter_update_thread_ = std::thread(update_parameters_fnc);
+  this->update_parameters_timer_ = this->create_wall_timer(this->period_, update_parameters_fnc);
```

Comment:
I think generally lambda functions are preferred over binding in modern usage. You would save yourself an intermediate variable too. Since these lines are being changed anyway, maybe you could update them here and in the other places (unless you disagree of course)
```suggestion
  this->update_parameters_timer_ = this->create_wall_timer(this->period_, [this] { this-> update_parameters(); });
```

Reply:
no sounds perfectly good. I didn't know about that syntax tbh